### ISSUE TITLE
About the role of training set in the process of prediction

### ISSUE THREAD

**Issue by jaegerstar** [openai/supervised-reptile#13]
>def _test_predictions(self, train_set, test_set, input_ph, predictions):
        if self._transductive:
            inputs, _ = zip(*test_set)
            return self.session.run(predictions, feed_dict={input_ph: inputs})
        res = []
        for test_sample in test_set:
            inputs, _ = zip(*train_set)
            inputs += (test_sample[0],)
            res.append(self.session.run(predictions, feed_dict={input_ph: inputs})[-1])
        return res

Why did you add train_set into _test_predictions function, since they had been learned in
>     self.session.run(minimize_op, feed_dict={input_ph: inputs, label_ph: labels})
which is 119th line of reptile.py

**Comment by unixpickle** [comment]
The reason is that, when you don't use transduction, you need to get BatchNorm statistics from somewhere. So, `_test_predictions` feeds the training inputs into the network as well as each test sample, in order to leverage the BatchNorm statistics from the training set.
